routing/simple.html

```diff
@@ -0,0 +1,240 @@
+      if (!baseurl)
+        baseurl = 'http://localhost:8002/';
+      baseurl += 'route?';
+      const token = document.getElementById('token').value;
+      if (token)
+        baseurl += 'access_token=' + token + '&json=';
+      else
+        baseurl += 'json=';
+      
+      //do them all
+      const responses = await Promise.all(
+        requests.map(async (request) => {
+          const url = baseurl + JSON.stringify(request);
+          const response = await $.getJSON(url);
+          return response;
+        })
+      );
+
+      //set the text box
+      return {requests, responses};
+    }
+
+    ////////////////////// RESPONSE PROCESSING /////////////////////////
+
+    //convert the result into geojson so its displayable
+    function build_geojson(responses) {
+      const fc = { type:'FeatureCollection', features: [] };
+
+      // for each response
+      responses.forEach( (response) => {
+        // pick a main color and then each alternate will be lighter
+        let color = Math.floor(Math.random()*16777215*.5);
+        let color_scale = 1.5;
+
+        // get the route geometry as linestirngs
+        const routes = [response.trip].concat(response.alternates || []);
+        routes.forEach( (route) => {
+          route.legs.forEach( (leg) => {
+            // the route
+            const route_feature = {
+              type: 'Feature',
+              properties: {
+                color: '#' + ('000000' + color.toString(16)).substr(-6),
+                opacity: 0.75,
+                weight: 5
+              },
+              geometry:{
+                type: 'LineString',
+                coordinates: decode(leg.shape, 1e6)
+              }
+            };
+
+            // the origin
+            const origin_feature = {
+              type: 'Feature',
+              properties: {
+                color: '#00bb00',
+                radius: 5,
+                fillOpacity: 0.75
+              },
+              geometry:{
+                type: 'Point',
+                coordinates: route_feature.geometry.coordinates[0]
+              }
+            }
+
+            // the destination
+            const destination_feature = {
+              type: 'Feature',
+              properties: {
+                color: '#bb0000',
+                radius: 5,
+                fillOpacity: 0.75
+              },
+              geometry:{
+                type: 'Point',
+                coordinates: route_feature.geometry.coordinates.slice(-1)[0]
+              }
+            };
+
+            fc.features.push(origin_feature);
+            fc.features.push(destination_feature);
+            fc.features.push(route_feature);
+          }); // legs
+
+          // alternates are lighter
+          color *= Math.floor(color_scale);
+          color_scale = 1;
+
+        }); // routes
+      }); // responses
+
+      //give back the feature collection
+      return fc;
+    }
+
+    function render(req_rep) {
```

Comment:
this updates the map with the geojson we created from the responses. it also updates the anchor/permalink and the textbox